packages/@okta/vuepress-site/docs/reference/hook-best-practices/index.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Hook best practices
+excerpt: >-
+  Understand the best practices at Okta to implement and secure an Okta Event Hook or Inline Hook.
+---
+
+# Hook best practices
+
+Event Hooks and Inline Hooks are outbound calls from Okta to an external service. These outbound calls integrate custom functionality into your Okta processes. For further background information on Okta Hooks, see [Event Hooks](/docs/concepts/event-hooks) and [Inline Hooks](/docs/concepts/inline-hooks).
+
+The following sections review best practices to implement and secure Okta Event Hooks or Inline Hooks.
+
+## Secure your Hook endpoint
+
+To prevent a malicious actor from making requests to the endpoint where your Okta Hooks are sent, use the following best practices:
+
+* Configure Okta to send an authentication header in the Hook and validate it in every request by:
+
+  * Using [HTTP Basic Authentication](/books/api-security/authn/api-authentication-options/#http-basic-authentication). When activating and enabling your Hooks on the Okta org, set the **Authorization field** as `authorization` and the **Authentication secret** in the Base64 `user:password` format.
+
+    >**Note:** You must include the authentication scheme as part of the **Authentication secret**. For Basic Authentication, your secret must appear similar to: `Basic Base64(user:password)`. See the following partial Hook header as an example:
+
+    ```cURL
+    'content-type',
+    'application/json',
+    'authorization',
+    'Basic YWRtaW46c3VwZXJzZWNyZXQ='
+    ```
+
+  * Using a separate Secret Key, which can be configured on the Hook as a custom header field.
+
+* Create an allowlist of IP addresses to check on incoming Okta calls. See the following Okta Knowledge Base article for a listing of Okta IP addresses: [What IP addresses/ranges should we allow for inbound traffic?](https://support.okta.com/help/s/article/What-IP-addresses-ranges-should-we-whitelist-for-inbound-traffic-i-e-REST-API-calls-from-Okta-to-on-prem-JIRA-server?language=en_US).
+
+>**Note:** This is a large list of IP Addresses and the list is subject to change. Unless required by your organization, securing your Hook by authentication header is recommended.
+
+## Protect your Hook content from external viewers
+
+To prevent unauthorized parties from reading the contents of an Okta Hook, we recommend the following:
```

Comment:
Okta recommends using HTTPS to encrypt communications to your Hook endpoint to prevent unauthorized parties from reading the contents of an Okta Hook. When using HTTPS, ensure you keep your SSL certificate updated and the Domain Name System (DNS) secured, so that someone can’t **reroute** your **calls** to another location.
(just have one paragraph)

Reply:
Thanks for your suggestion. Looks great. Although, I believe it's not a recommendation, but a requirement. I don't think Hooks will work with out HTTPS. But let me double-check that.

Reply:
Confirmed with PM (and our management API docs [Create Event Hooks](https://developer.okta.com/docs/reference/api/event-hooks/#create-event-hook) that HTTPS is a requirement. Re-worded opening to: "Okta requires HTTPS to encrypt communications to ...."

Also added a note:

>**Note:** If you add an HTTP URL when enabling a Hook in the Admin Console, an Invalid URL provided error displays.